File: src/web/magmac/app/lang/InvocationFolder.ts
import { DivideState } from "../../../magmac/app/compile/rule/divide/DivideState";
import { Folder } from "../../../magmac/app/compile/rule/fold/Folder";
export class InvocationFolder {
	fold(state : DivideState, c : char) : DivideState {appended : var=state.append( c);if(true){ return appended.enter( ).advance( );;}if(true){ return appended.enter( );;}if(true){ return appended.exit( );;}return appended;;}
	createDelimiter() : String {return "";;}
}

File: src/java/magmac/app/lang/InvocationFolder.java
package magmac.app.lang;

import magmac.app.compile.rule.divide.DivideState;
import magmac.app.compile.rule.fold.Folder;

public class InvocationFolder implements Folder {
    @Override
    public DivideState fold(DivideState state, char c) {
        DivideState appended = state.append(c);
        if ('(' == c && appended.isLevel()) {
            return appended.advance();
        }
        if ('(' == c) {
            return appended.enter();
        }
        if (')' == c) {
            return appended.exit();
        }
        return appended;
    }

    @Override
    public String createDelimiter() {
        return "";
    }
}
